src/libs/tmdk/src/commands/deploy.ts

```diff
@@ -130,12 +144,15 @@ export const handler = async (argv: Arguments<Options>) => {
               `Created component-type: ${componentTypeDefinition["componentTypeId"]}`
             );
           }
-        } catch (error) {
-          if (error instanceof ValidationException) {
-            // TODO check message is due to something not existing in workspace
+        } catch (e) {
+          if (
+            // components are not gaurenteed to be in the correct order; retry
+            e instanceof ValidationException &&
+            e.message.includes("do not exist in workspace")
```

Comment:
Where is this error message from? Seems long and specific

Reply:
This is the fix to one of the convos in the slack channel about Johnny's athena connector component type causing a hang (infinite loop). Previously we caught all validation errors and assumed that it was a "does not exist in workspace" error, so other types of errors would be included in this.

This change allows for the specific, expected error to be caught and so we continue with component type creation, while throwing and stopping execution at any other unexpected error.


In the future, we may want to change the entire component type creation logic (one of the TODO's is "cleaner dependency creation process" but for now, it brute force tries to create them all in a loop. If they are incorrectly ordered, it throws this "does not exist" error, moves on, and then eventually revisits the failed components. There may be a way to determine the dependencies beforehand and reorder the component types instead.

Reply:
That makes sense but where did this message come from? Is this recorded from an api call you made or documented somewhere in the code? My concern is that the error message is actually 'does not exist' and so 'do not exist' will not catch the exception

Reply:
Oh I see your concern, that's my bad I probably should have included a longer snippet of the error message. It uses .includes so there was no need to include the full message.

Here is a longer snippet from one of my test runs on a workspace named SyncB:
"ValidationException: Component types [com.example.timestream-telemetry] do not exist in workspace SyncB."

Unfortunately the API documentation does not provide a list of all Validation errors: https://docs.aws.amazon.com/iot-twinmaker/latest/apireference/API_CreateComponentType.html#tm-CreateComponentType-request-tags

Reply:
Since the "SyncB" is workspace name and the "[com.example.timestream-telemetry]" is name of the component, there actually is no better way for my to include an clearer error message.

Reply:
Got it, makes sense. I am thinking we can just check if the exception is validation exception and remove the error message check to simplify. Is there a validation exception we would not want to retry on? Up to you though, resolve this thread if you feel it is fine as is

Reply:
The previous version was just the validation exception, but the error message was necessary because there might be unexpected errors (like the one Johnny brought up in the slack channel) that would cause infinite loop.